src/custom-element.js

```diff
@@ -785,6 +792,8 @@ export function createAmpElementProto(win, name, opt_implementationClass) {
     this.assertNotTemplate_();
     dev.assert(this.isUpgraded() && this.isBuilt(),
         'Must be upgraded and built to receive viewport events');
+    // Reset hadLoadingError_ when trying to load resources.
+    this.hadLoadingError_ = false;
```

Comment:
I don't think this is the right location. `#layoutCallback` should only be called once (whether is succeeds or fails), until `#unlayoutCallback` is called. I think setting this in `#unlayoutCallback` with the `layoutCount = 0` is the right spot.